server/index.js

```diff
@@ -72,6 +82,11 @@ postload.forEach((middleware) => app.use(middleware))
 
 // error handler for rendering the 404 and 500 pages, must go last
 app.use(errorPages)
-app.listen(parseInt(process.env.PORT || '3000', 10))
+
+// If we are testing, we don't need to listen on port 3000
```

Comment:
Instead of switching off the `test` NODE_ENV here, could we switch off of `require.main === module` ? Then we would always listen on the port when this file is run directly, but not start listening when the server is imported by another file (like for tests).